sdk/src/sharedTest/java/io/customer/sdk/queue/QueueStorageIntegrationTest.kt

```diff
@@ -46,52 +47,57 @@ class QueueStorageIntegrationTest : BaseTest() {
 
     @Test
     fun givenCreateTask_expectDoesNotReturnTaskWhenAskingForDifferentId() {
-        queueStorage.create("foo", "foo")
+        queueStorage.create("foo", "foo", null, null)
         queueStorage.get("123").shouldBeNull()
     }
 
     @Test
     fun givenTaskCreated_expectToGetTask() {
         val givenType = String.random
         val givenData = String.random
+        val givenGroupStart = QueueTaskGroup.IdentifyProfile("148")
+        val givenGroupMembers = listOf(QueueTaskGroup.RegisterPushToken("ABC"))
 
-        queueStorage.create(givenType, givenData)
-        val newlyCreatedTaskId = queueStorage.getInventory()[0].taskPersistedId
+        queueStorage.create(givenType, givenData, givenGroupStart, givenGroupMembers)
+        val newlyCreatedInventoryItem = queueStorage.getInventory()[0]
+        val createdTask = queueStorage.get(newlyCreatedInventoryItem.taskPersistedId)
 
-        val createdTask = queueStorage.get(newlyCreatedTaskId)
+        newlyCreatedInventoryItem.shouldNotBeNull()
         createdTask.shouldNotBeNull()
 
         createdTask.type shouldBeEqualTo givenType
         createdTask.data shouldBeEqualTo givenData
+        newlyCreatedInventoryItem.groupStart shouldBeEqualTo "identified_profile_148"
+        newlyCreatedInventoryItem.groupMember shouldBeEqualTo listOf("registered_push_token_ABC")
     }
 
     @Test
     fun givenCreateNewTask_expectInventoryUpdated() {
         queueStorage.getInventory().isEmpty().shouldBeTrue()
 
-        queueStorage.create(String.random, String.random)
+        queueStorage.create(String.random, String.random, null, null)
```

Comment:
A bit general question, related to the `blocking group` concept. 

Are we doing anything to avoid the cyclic dependency? both groups being blocked by each other?

Reply:
Ah! That would indeed be a big problem if that happened. 

With all of the apps that I have built using a background queue, and our specific use cases of our SDK, I have not encountered a scenario where cyclic dependencies happened. However, a bug could easily cause this to happen when we didn't want it to happen. 

On the SDK level, we are currently not doing anything to avoid this from happening. Off the top of my head, some ideas for how we could address this concern is:
* We could write a series of automated tests that check if cyclic dependencies are possible. 
* We create and maintain a flow chart graph (like the one below) for all groups that are in the SDK to help us visualize our dependencies to try and catch cyclic dependencies. 

<img width="927" alt="CleanShot 2022-04-13 at 15 58 33@2x" src="https://user-images.githubusercontent.com/2041082/163269284-5c405762-e0e0-4bc2-856f-8905f048deb6.png">

* We could write some code to catch cyclic dependencies in the SDK. But that solution has some drawbacks. For one, that code could also have a bug in it. Also, that code exists as more of a band-aid to cover up the bug after it already exists. I think that if we have preventative measures in place during development to avoid introducing cyclic dependencies in the first place, we wouldn't need this logic. 

I agree this is something important to think about. Cyclic dependencies could cause problems that prevent a lot of data from being sent to CIO. Any ideas on how to prevent it?

Reply:
On the top of my head, we could use `LinkedList` instead of `List` and find a loop in it. 
This can be done while adding a task to the queue.

Reply:
Could you elaborate on "find a loop in it"? I can see how a `LinkedList` would be a good data structure for this scenario but I know the data structure by itself would not solve the problem. 

Want to create a ticket and you and I share ideas together there? I find this idea to be a good improvement to the background queue. I also do not believe this to be a blocker for this PR so a new ticket will help us keep the conversation going.

Reply:
Yup, not a blocker, I'll create a ticket and propose a `LinkedList` based solution, which upon addition of any task would traverse itself to find a possible `loop/circular depdency`.  👍 

This conversation can be closed.

Reply:
Created a ticker for it, In there I'll propose how I believe `LinkedList` can be utilized, that every time a new task is added, it's traversed to find a possible `loop/circular dependency` in it. 

This conversation can be closed as well.